sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzurePath.java

```diff
@@ -58,182 +119,422 @@ public Path getRoot() {
      */
     @Override
     public Path getFileName() {
-        return null;
+        if (this.withoutRoot().isEmpty()) {
+            return null;
+        } else {
+            return this.parentFileSystem.getPath(Flux.fromArray(this.splitToElements()).last().block());
```

Comment:
Any reason to use a reactive stream here only to block it? Instead could a `Stream` be used? Another option, based on the logic here, we could have a local instance of the `String[]` and just access and return the last element in the list.

Reply:
I think I addressed the Stream concern above and the String[] concern below :)